src/methods/michelsen/saturation_pure.jl

```diff
@@ -53,12 +51,12 @@ function psat_fugacity(model::EoSModel, T, p0, vol0=[nothing, nothing])
     # vol0 = initial guesses for the phase volumes = [vol liquid, vol vapor]
     # out = Saturation Pressure, vol liquid, vol vapor
     vol_liq0, vol_vap0 = vol0
-
+    z = SA[1.]
     RT = R̄*T
     P = 1. * p0
     # Solving the phase volumes for the first iteration
-    vol_liq = volume(model, P, T, phase=:liquid, vol0=vol_liq0)
-    vol_vap = volume(model, P, T, phase=:vapor, vol0=vol_vap0)
+    vol_liq = _volume_compress(model, P, T, z, x0_volume_liquid(model,T,z))
```

Comment:
Hey Andres!

Thanks for all the work yo've done in such a short time! I have one concern with this modification, as it is obtaining the initial guess automatically it would prevent using an user supplied initial guess  ` vol_liq0, vol_vap0 = vol0` (if were the case)

Gustavo.

Reply:
I see hahaha, i forgot that part. on the last PR is initialized correctly.
Now that we have the methods, we need to integrate all those into the Clapeyron.jl function ecosystem. in Julia there is normally this style of `solve(problems::Problem,x0::X0,method::Method)` so we can dispatch to efficient implementations of `solve` for specific combinations of problems types, initial point types and method types. at the moment, we only have this implemented on `tp_flash` (for example,`tp_flash(model,p,T,z,method::TPFlashMethod)`, 
but a similar api for `saturation_pressure`, `saturation_pressure` and `volume` is missing. our (default) volume solver is just a thin wrapper to the iterative volume solver with support for threaded evaluation (liquid and gas branches at the same time). 
for this PR to be merged, we need to:

- [ ] remove the `Optim` and  `NLsolve ` dependencies.  `NLSolvers.jl ` has a really sparse documentation hahaha, but it excels at small sizes (via static vector support). i already made a port of `only_fg!`, `only_fgh!` and `only_fj!` to be used by NLSolvers.jl, so hopefully, the switch can go without errors.
- [ ] move `psat` and `tsat` to a method interface, integrate with `saturation_pressure` and `saturation_temperature`: we need to define some types (mainly `SaturationPressureMethod` and `SaturationTemperatureMethod`) and make  `saturation_pressure` and `saturation_temperature` use those. 
- [ ] merge duplicated code. we already have a `RRTPFlash` method for only rachford rice, and a richford rice solver, so we need to merge those to be usable. ideally the SS part of michelsen could be done by making a invoking an RR tp flash and then refining via gibbs minimization.
- [ ] address wilson + SAFTgammaMie:  weird and i have to check if i can reproduce, but i imagine some points of failure
- [ ] documentation: add doc strings on every new method, add those to the the doc page
- [ ] tests: test everything @pw0908 do we use the same test system?

Reply:
Thanks a lot Andrés!

Sorry my ignorance (this is my first PR ever), but for the checklist that you mention, is there something I could do to help? I don't seem to find the option to sync your modifications with my fork e.e

Gustavo

Reply:
Hi Gustavo! Regarding syncing: Andrès pushed the commits directly into your fork (as I believe you have given us permissions to push to your fork when you made the PR). We typically continue to make changes on the branch that the PR was created from until all open issues are resolved, so that the changes can be merged to our repo. You should be able receive all of these modifications by doing a `git pull` from the terminal, but there should be some sort of button in whichever editor that you’re using that does the same thing.